chore: migrate away from Node's EventEmitter

jackfranklin opened the pull request:
This PR re-introduces Mitt (that was released in v3.2 and reverted in v3.3).

The majority of EventEmitter methods are re-implemented in Puppeteer and we will release this as a major version accompanied by release notes that document exactly which methods have been removed to help users migrate.

jackfranklin committed:
bring back initial mitt

jackfranklin committed:
reimplementing mitt with tests

jackfranklin committed:
Cover our event emitter in tests

jackfranklin committed:
add spec for on + off and request handling

jackfranklin commented on docs/api.md:
```diff
@@ -3935,6 +3942,35 @@ reported.
+- `event` <[string]|[symbol]> the event to remove the handler from.
+- `handler` <[Function]> the event listener that will be removed.
+
+This method is identical to `off` and maintained for compatibility with Node's EventEmitter. We recommend using `off` by default.
```
these docs are not up to date and need to be written again.

mathiasbynens reviewed (approved):
LGTM % the TODO you outlined

jackfranklin commented on test/page.spec.js:
```diff
@@ -130,6 +131,25 @@ describe('Page', function () {
+      // Two now because we added the handler back.
+      expect(handler.callCount).toBe(2);
+    });
+  });
```
a user reported a bug in 3.2 where this code didn't work. I don't think it was a bug in Mitt, but I wanted to add a test that explicitly listened to an event, stopped, and then started again just to make sure. It's a useful test to have IMO.

jackfranklin committed:
Fix the docs for the new event emitter

jackfranklin committed:
Remove mentions and links of Node's `EventEmitter`

And link to our event emitter instead. This ensures that we don't send
anyone erroneously to the NodeJS docs.

jackfranklin committed:
document the new event listener

jackfranklin committed:
link to event emitter

jackfranklin committed:
skip firefox test

jackfranklin committed:
try enabling ff test

jackfranklin committed:
ff definitely needs the skip

jackfranklin committed:
Do not document this interface
